src/qgis_geonode/gui/geonode_source_select_provider.py

```diff
@@ -61,99 +40,65 @@ def ordering(self):
         return QgsSourceSelectProvider.OrderOtherProvider
 
 
-class CustomGeonodeWidget(QgsAbstractDataSourceWidget, WidgetUi):
+class GeonodeDataSourceWidget(QgsAbstractDataSourceWidget, WidgetUi):
     def __init__(self, parent, fl, widgetMode):
-        super(CustomGeonodeWidget, self).__init__(parent, fl, widgetMode)
+        super().__init__(parent, fl, widgetMode)
         self.setupUi(self)
         self.project = QgsProject.instance()
         self.settings = QgsSettings()
-
+        self.connections_cmb.currentIndexChanged.connect(
+            self.toggle_connection_management_buttons
+        )
         self.btnNew.clicked.connect(self.add_connection)
         self.btnEdit.clicked.connect(self.edit_connection)
         self.btnDelete.clicked.connect(self.delete_connection)
+        self.toggle_connection_management_buttons()
 
-        # Update GUI
-
-        self.create_connections_list()
+        settings_manager.current_connection_changed.connect(
+            self.update_connections_combobox
+        )
```

Comment:
we might need to update the combo box connection here, currently connections that were added, don't show up when the widget is loaded for the first time.